lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/Watcher.scala

```diff
@@ -0,0 +1,42 @@
+package org.enso.filewatcher
+
+import java.nio.file.Path
+
+trait Watcher {
```

Comment:
This PR is a nice opportunity to separate API (e.g. `Watcher`) from the implementation and have them in different modules.